### PR TITLE
Add numpy, polars and pandas as extra dependencies

### LINKED ISSUE #90: Enable `extras` for installation
Currently, if somebody wants to use the pandas or numpy functionality, they need to manually install those packages. This is fine, but conventionally there is the ability to install extra required packages with the extras syntax, ie: `pip install pycsvy[pandas]`

This can be added to the pyproject.toml using [`extras`](https://python-poetry.org/docs/pyproject/#extras)

### PR DESCRIPTION
**Pull request by dalonsoa**
Such that, eventually, the chosen engine can be installed with `pip install pycsvy[pandas]`, for example. 

It also removes the `optional` dependencies for the groups since using poetry directly is meant for developers and they will want, normally, all the dependencies. 

Close #90

### CODE REVIEW

File: .github/workflows/ci_template.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         poetry-version: 1.8.3
 
     - name: Install dependencies
-      run: poetry install --with dev
+      run: poetry install
```
Comment (dc2917) [inline .github/workflows/ci_template.yml:35]: So this installs optional dependencies by default?

**Review by dc2917 — APPROVED** [review]

**Review by AdrianDAlessandro — APPROVED** [review]
LGTM! I assume this works with the `pip install pycsvy[pandas]` notation because it follows the docs. Hard to work out how to test it though.

File: .github/workflows/ci_template.yml (hunk 1 above)
Comment (dalonsoa) [reply .github/workflows/ci_template.yml:35]: Yep. Now, if you do NOT want them, you do, for example, `poetry install --without dev`, but the default position is install all dependencies. As mentioned above, poetry is for developers, not for end users, and they will typically want all the pedendencies.

Comment (AdrianDAlessandro) [reply .github/workflows/ci_template.yml:35]: Not exactly. Poetry installs all groups by default except for `extras` and those marked as optional. Previously, the `dev` and `docs` groups were marked as optional, but this PR changes that

**Comment by dalonsoa** [comment]
I guess you can create a wheel with `poetry build`, then, in a clear, brand new environment - can be a `venv` one - run `pip install path_to_wheel[pandas]`. Maybe, haven't tried.

**Comment by dalonsoa** [comment]
Just checked what I said and it works as planned: `python -m pip install .\dist\pycsvy-0.2.2-py3-none-any.whl[pandas]` installs pycsvy with pandas.

**Pull request merged**